presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -262,6 +274,8 @@ public static ConnectorPageSource createParquetPageSource(
                     hiveFileContext.incrementCounter("parquet.blocksSkipped", 1);
                     hiveFileContext.incrementCounter("parquet.rowsSkipped", block.getRowCount());
                     hiveFileContext.incrementCounter("parquet.totalBytesSkipped", block.getTotalByteSize());
+                    //what if columnIndexStore is null
```

Comment:
did not get it. if the predicate does no match, why we add `columnIndexStore` into `blockIndexStores`?

Reply:
Good catch! Last rebase moved it. Let's fix it.